### PR TITLE
Support 'trace.xxx' filter string in search_traces() API

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/12193?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12193/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12193
```

</p>
</details>

### What changes are proposed in this pull request?

In search traces API, `trace.xxx` should work as alias for `attribute.xxx`.

**Note**: I don't see any documentation mentioning this alias. I will file a follow-up to enrich for docstring for `search_traces()` API. 

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

Verified below in both file and SQL backend.

##### Search with `trace.status`.
Screenshot 2024-05-31 at 14 10 18

#### Search with invalid entity
Screenshot 2024-05-31 at 14 11 08

#### Search with invalid attribute
Screenshot 2024-05-31 at 14 11 15

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/store/tracking/test_file_store.py [hunk 1]
```diff
@@ -3034,6 +3048,27 @@ def test_search_traces_filter(generate_trace_infos):
...
 
 
+@pytest.mark.parametrize(
+    ("filter_string", "error"),
+    [
+        ("invalid", r"Invalid clause\(s\) in filter string"),
+        ("name = 'foo' AND invalid", r"Invalid clause\(s\) in filter string"),
+        ("foo.bar = 'baz'", r"Invalid entity type 'foo'"),
+        ("invalid = 'foo'", r"Invalid attribute key 'invalid'"),
+        ("trace.tags.foo = 'bar'", r"Invalid attribute key 'tags.foo'"),
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:3058]:
```suggestion
        ("trace.tags.foo = 'bar'", r"Invalid attribute key 'tags\.foo'"),
```

File: mlflow/utils/search_utils.py [hunk 2]
```diff
@@ -1600,6 +1612,19 @@ def is_request_metadata(cls, key_type, comparator):
...
+        entity_type = cls._trim_backticks(entity_type)
+        if entity_type not in cls._VALID_IDENTIFIERS:
+            raise MlflowException(
+                f"Invalid entity type '{entity_type}'. Valid values are {cls._VALID_IDENTIFIERS}",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        elif entity_type in cls._ALTERNATE_IDENTIFIERS:
+            return cls._ALTERNATE_IDENTIFIERS[entity_type]
+        else:
+            return entity_type
```
Comment (harupy) [inline mlflow/utils/search_utils.py:1626]:
```suggestion
        return cls._ALTERNATE_IDENTIFIERS.get(entity_type, entity_type)
```

does this work?

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/utils/search_utils.py (hunk 2 above)
Comment (B-Step62) [reply mlflow/utils/search_utils.py:1626]: Yup, but maybe less readable?

File: tests/store/tracking/test_file_store.py (hunk 1 above)
Comment (B-Step62) [reply tests/store/tracking/test_file_store.py:3058]: wow what a coincidence this passed lol🙃

**Comment by B-Step62** [comment]
The failed test is not related to the change
> FAILED | MEM 2.4/15.6 GB | DISK 48.5/72.5 GB tests/pyfunc/docker/test_docker.py::test_build_image[params7] - RuntimeError: Docker build failed.
> 143.3 E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/x/xft/libxft2_2.3.3-0ubuntu1_amd64.deb  Undetermined Error [IP: 185.125.190.36 80]
> 143.3 E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/libx/libxmu/libxmu6_1.1.3-0ubuntu1_amd64.deb  Undetermined Error [IP: 185.125.190.36 80]

**Pull request merged**